pkg/controller/azvolume.go

```diff
@@ -293,6 +348,40 @@ func (r *reconcileAzVolume) DeleteAzVolume(ctx context.Context, azVolume *v1alph
 	return nil
 }
 
+func (r *reconcileAzVolume) CleanUpAzVolumeAttachment(ctx context.Context, azVolumeName string) error {
+	var azVolume v1alpha1.AzVolume
+	err := r.client.Get(ctx, types.NamespacedName{Namespace: r.namespace, Name: azVolumeName}, &azVolume)
+	if err != nil {
+		klog.Errorf("failed to get AzVolume (%s): %v", azVolumeName, err)
+		return err
+	}
+
+	volRequirement, _ := labels.NewRequirement(VolumeNameLabel, selection.Equals, []string{azVolume.Spec.UnderlyingVolume})
+	labelSelector := labels.NewSelector().Add(*volRequirement)
+
+	attachments, err := r.azVolumeClient.DiskV1alpha1().AzVolumeAttachments(r.namespace).List(ctx, metav1.ListOptions{LabelSelector: labelSelector.String()})
+	if err != nil && !errors.IsNotFound(err) {
+		klog.Errorf("failed to get AzVolumeAttachments: %v", err)
+		return err
+	}
+
+	for _, attachment := range attachments.Items {
+		if err = r.azVolumeClient.DiskV1alpha1().AzVolumeAttachments(r.namespace).Delete(ctx, attachment.Name, metav1.DeleteOptions{}); err != nil {
+			klog.Errorf("failed to delete AzVolumeAttachment (%s): %v", attachment.Name, err)
+			return err
+		}
+		klog.V(5).Infof("Set deletion timestamp for AzVolumeAttachment (%s)", attachment.Name)
+	}
+
+	if err := r.UpdateStatus(ctx, azVolumeName, v1alpha1.VolumeReleased, false); err != nil {
```

Comment:
Isn't the status of AzVolume already set to Released?

Reply:
Aya, I meant `VolumeAvailable`. Thank you for pointing this out.